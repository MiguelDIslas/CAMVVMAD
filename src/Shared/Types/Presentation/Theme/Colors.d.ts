interface Colors {
  /**
   * Color principal de la aplicación
   */
  primary: string;

  /**
   * Color secundario de la aplicación
   */
  secondary: string;

  /**
   * Color blanco
   */
  white: string;

  /**
   * Color negro
   */
  black: string;

  /**
   * Color tono obscuro
   */
  darker: string;

  /**
   * Color gris
   */
  gray: string;

  /**
   * Color gris claro
   */
  lightGray: string;

  /**
   * Color naranja
   */
  orange: string;
}
