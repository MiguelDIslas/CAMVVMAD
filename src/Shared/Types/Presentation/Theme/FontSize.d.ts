interface FontSize {
  /**
   * Tamaño de fuente extra pequeño (12px)
   */
  xs: string;

  /**
   * Tamaño de fuente pequeño (14px)
   */
  s: string;

  /**
   * Tamaño de fuente medio (16px)
   */
  m: string;

  /**
   * Tamaño de fuente grande (18px)
   */
  l: string;

  /**
   * Tamaño de fuente extra grande (20px)
   */
  xl: string;
}
