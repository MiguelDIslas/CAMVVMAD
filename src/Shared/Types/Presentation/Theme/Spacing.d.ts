interface Spacing {
  /**
   * Tamaño de espaciado extra pequeño (4px)
   */
  xs: string;

  /**
   * Tamaño de espaciado pequeño (8px)
   */
  s: string;

  /**
   * Tamaño de espaciado medio (16px)
   */
  m: string;

  /**
   * Tamaño de espaciado grande (24px)
   */
  l: string;

  /**
   * Tamaño de espaciado extra grande (32px)
   */
  xl: string;
}
