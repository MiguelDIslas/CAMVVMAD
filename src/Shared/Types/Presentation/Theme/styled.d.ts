//Definición Principal para el tema de la aplicación en styled-components
import 'styled-components/native';

declare module 'styled-components/native' {
  export interface ThemeArgument {
    /**
     * Colores de la aplicación
     */
    colors: Colors;

    /**
     * Tamaño de espacios
     */
    spacing: Spacing;

    /**
     * Tamaño para textos
     */
    fontSize: FontSize;
  }

  export interface DefaultTheme extends ThemeArgument {}
}
