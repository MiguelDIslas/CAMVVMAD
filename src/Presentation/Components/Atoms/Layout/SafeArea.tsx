import {SafeAreaView, StatusBar} from 'react-native';
import styled from 'styled-components/native';
import {isAndroid} from '@Helpers/Platform.helper';

type SafeAreaProps = {
  /**
   * Color de fondo del SafeAreaView
   */
  backgroundColor?: string;
};

/**
 * Componente que renderiza un SafeAreaView con un color de fondo personalizado. 
 * Si es Android, se toma en cuenta el StatusBar.
 */
const SafeArea = styled(SafeAreaView)<SafeAreaProps>`
  flex: 1;
  background-color: ${props =>
    props.backgroundColor || props.theme.colors.darker};
  padding-top: ${isAndroid ? StatusBar.currentHeight : 0}px;
`;

export default SafeArea;
