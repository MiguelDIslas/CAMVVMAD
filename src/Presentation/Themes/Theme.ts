import {ThemeArgument} from 'styled-components/native';
import {colors} from './Colors';
import {spacing} from './Spacing';
import {fontSize} from './FontSize';

const myTheme: ThemeArgument = {
  colors,
  spacing,
  fontSize,
};

export {myTheme};
