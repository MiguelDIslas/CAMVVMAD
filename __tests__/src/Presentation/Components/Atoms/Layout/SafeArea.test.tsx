import React from 'react';
import {render} from '@testing-library/react-native';
import {ThemeProvider} from 'styled-components/native';

import {myTheme} from '@Themes/Theme';
import SafeArea from '@Components/Atoms/Layout/SafeArea';

describe('SafeArea', () => {
  it('Debe retornar los colores por defecto, un background obscuro y texto claro', () => {
    const {toJSON} = render(
      <ThemeProvider theme={myTheme}>
        <SafeArea testID="safe-area" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
    expect(toJSON()).toHaveStyle({backgroundColor: myTheme.colors.darker});
  });


  it('Debe retornar los colores personalizados, un background claro y texto obscuro', () => {
    const {toJSON} = render(
      <ThemeProvider theme={myTheme}>
        <SafeArea
          testID="safe-area"
          backgroundColor={myTheme.colors.orange}
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
    expect(toJSON()).toHaveStyle({backgroundColor: myTheme.colors.orange});
  });

});
