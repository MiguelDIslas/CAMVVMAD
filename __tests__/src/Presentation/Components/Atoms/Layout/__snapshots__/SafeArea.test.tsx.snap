// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeArea Debe retornar los colores personalizados, un background claro y texto obscuro 1`] = `
<RCTSafeAreaView
  backgroundColor="#E16428"
  style={
    {
      "backgroundColor": "#E16428",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingTop": 0,
    }
  }
  testID="safe-area"
/>
`;

exports[`SafeArea Debe retornar los colores por defecto, un background obscuro y texto claro 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#272121",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingTop": 0,
    }
  }
  testID="safe-area"
/>
`;
